src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SupportApiServiceImpl.scala

```diff
@@ -0,0 +1,196 @@
+import org.opensaml.xml.validation.ValidationException
+import org.opensaml.xml.{XMLObject, Configuration}
+import org.opensaml.DefaultBootstrap
+import org.opensaml.xml.io.{Unmarshaller, UnmarshallerFactory}
+
+import cats.syntax.all._
+import java.time.{OffsetDateTime, Instant, ZoneOffset}
+import java.io.ByteArrayInputStream
+import scala.jdk.CollectionConverters._
+import scala.concurrent.{Future, ExecutionContext}
+import scala.util.Success
+
+final case class SupportApiServiceImpl(
+  sessionTokenGenerator: SessionTokenGenerator,
+  tenantProcessService: TenantProcessService,
+  offsetDateTimeSupplier: OffsetDateTimeSupplier
+)(implicit ec: ExecutionContext)
+    extends SupportApiService {
+
+  private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  val SUPPORT_LEVELS: Seq[String] = Seq("L2", "L3")
+  val SUPPORT_LEVEL_NAME: String  = "supportLevel"
+  val OPERATOR: String            = "OPERATOR"
+
+  override def redirectToSupportPage(
+    sAMLResponse: SAMLResponse
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+
+    logger.info(s"Calling Support SAML")
+
+    val result: Future[(String, String)] = for {
+      _            <- validate(parseResponse(sAMLResponse.response))
+      tenant       <- ApplicationConfiguration.pagoPaTenantId.toFutureUUID >>= tenantProcessService.getTenant
+      selfcareId   <- tenant.selfcareId.toFuture(MissingSelfcareId(tenant.id))
+      sessionToken <- sessionTokenGenerator.generate(
+        signatureAlgorithm = SignatureAlgorithm.RSAPkcs1Sha256,
+        claimsSet = buildClaims(selfcareId, tenant),
+        audience = ApplicationConfiguration.generatedJwtAudience,
+        tokenIssuer = ApplicationConfiguration.generatedJwtIssuer,
+        validityDurationInSeconds = 300
+      )
+      base64       <- sAMLResponse.response.encodeBase64.toFuture
+    } yield (base64, sessionToken)
+
+    onComplete(result) {
+      handleError(s"Error calling support SAML") orElse { case Success((base64, sessionToken)) =>
+        val redirectUrl =
+          s"https://selfcare.dev.interop.pagopa.it/ui/it/assistenza/scelta-ente#saml2=$base64&jwt=$sessionToken"
+        redirect(redirectUrl, StatusCodes.MovedPermanently)
+      }
+    }
+  }
+
+  override def getSaml2Token(tenantId: String, sAMLResponse: SAMLResponse)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerSessionToken: ToEntityMarshaller[SessionToken],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Calling get SAML2 token")
+
+    val result: Future[SessionToken] = for {
+      _            <- validate(parseResponse(sAMLResponse.response))
+      tenant       <- tenantId.toFutureUUID >>= tenantProcessService.getTenant
+      selfcareId   <- tenant.selfcareId.toFuture(MissingSelfcareId(tenant.id))
```

Comment:
@beetlecrunch not sure if this may be a requirement, but this implies that support cannot log in as an organization that has never accessed the platform.
We could add a specific error message for this case?

Reply:
A specific error message it's useful.

Reply:
it is, but it is not handled in the `handleError`, and the reported status is 500 with a generic message

Reply:
It is a wanted behaviour, but if you want I can catch this error and trasform it to a BadRequest  error

Reply:
considering 
> the endpoints that list institutions will show only tenants with selfcareId

and that this is an internal endpoint, we can leave it as 500 for me